config/zephyr/Kconfig

```diff
@@ -477,7 +477,19 @@ config CHIP_FACTORY_RESET_ERASE_NVS
 	  regain the original storage performance if any firmware issue has brought
 	  it to an unexpected state. For this reason, set this option if the entire
 	  configuration is supposed to be cleared on a factory reset, including
-	  device-specific entries.
+	  device-specific entries. It makes sense only for NVS filesystem.
+
+config CHIP_FACTORY_RESET_ERASE_ZMS
```

Comment:
It's fine for now and will accept this, however, what do you think to have one Matter config like `CHIP_FACTORY_RESET_PARTITION_ERASE` which would use ZMS or NVS underneath based on the selected backend?

Reply:
Yeah but it impacts many platforms that use Zephyr... We can do it, but not so easily. :/ I think we can create a follow up task for that after the release. :)

Reply:
Yeah, I also think that using two different Kconfigs for that is not the best idea. Why is it not that easy? Isn't it just about changing CHIP_FACTORY_RESET_ERASE_NVS->CHIP_FACTORY_RESET_ERASE_SETTINGS and then adding depends on depends on SETTINGS_NVS || SETTINGS_ZMS? Then we can use zms_clear or nvs_clear in the cpp file based on SETTINGS_NVS/SETTINGS_ZMS selection, because it's a choice. I don't think it would affect anyone much unless he will use SETTINGS_ZMS.